### ISSUE TITLE
Issue ending staking after using flashDeposit function

### ISSUE THREAD

**Issue by UPLukeet** [gemworks/gem-farm#74]
I have staked some NFTs using the flashDeposit but every time I try and call the end staking function I receive this error `Uncaught (in promise) Error: failed to send transaction: Transaction simulation failed: Error processing Instruction 0: custom program error: 0x1770`
I'm unsure why this is the vault appears locked on your front-end and even trying to end staking there doesn't work. I'm kind of stuck atm.

**Comment by UPLukeet** [comment]
![image](https://user-images.githubusercontent.com/57526569/163193144-69258692-5d55-4f0f-a91b-e052b5823f2e.png)

**Comment by Nazaire** [comment]
This seems to be caused by the "stakedFarmerCount" value. 
My theory is:
- flash deposit doesn't seem to increment stakedFarmerCount properly. It was showing as zero when I definitely had 1 staked account.
- When you hit end staking, it'll try and decrement the total staked accounts. Subtracting 1 from 0, throwing an error. Not allowing the farmer to end staking.

What I did to fix it: (I wouldn't call this fixing..)
- Create another farmer / vault, deposit a gem, and stake the vault so the "stakedFarmerCount" is at least 1.
- Then your original vault will be able to end stake...
- Now you'll see this error with the new farmer.

So just make sure you're not lucky last...

**Comment by Nazaire** [comment]
<img width="1043" alt="CleanShot 2022-04-17 at 16 16 11@2x" src="https://user-images.githubusercontent.com/2215998/163703087-cb3160d5-951d-4133-8291-adba818cd6d5.png">

For example here is my current farm `fj872SB4HiQ5VcuxXYrwVxoxvkghsaP5wfSmS5JEELS` on devnet that is showing zero for zero for `stakedFarmerCount` even though there is 4 gems staked. Which shouldn't be possible.

**Comment by ArthurPaivaT** [comment]
Hey! I had the same issue when I deployed my own version, what you can do to overcome this issue is send a startStaking instruction alng with the flashDeposit one, this way the counter will increment and you will be able to end staking later. It will bug the counter and if you need to know how many nfts are staked, fetch all the gems deposit PDAs. 

If you are deploying your own version, you can edit the code to increment the counter in the flashDeposit instruction.

TLDR: Send some start staking transactions before.

**Comment by enzoampil** [comment]
@ArthurPaivaT can you link to this `startStaking` instruction?

**Comment by ilmoi** [comment]
Thanks for raising the issue guys.

The problem arises because `flashDeposit` was never meant to be used before `stake` instruction, and so it rightly assumes the farmer is already staked and doesn't increment the count. My bad for not realizing it could be used on its own.

[Latest commit](https://github.com/gemworks/gem-farm/commit/23ae74bcc981b59228371a9ce434f3fc6ab88bd7) fixed this. Thx.
